### PR TITLE
Remove unneeded "replace" directives.

### PR DESCRIPTION
**Pull request by cgruber**

### CODE REVIEW

File: deployment/go.mod [hunk 1]
```diff
@@ -521,7 +521,5 @@ require (
 replace (
 	// replicating the replace directive on cosmos SDK
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
-	github.com/sourcegraph/sourcegraph => github.com/sourcegraph/sourcegraph-public-snapshot v0.0.0-20240822153003-c864f15af264
-
-	github.com/sourcegraph/sourcegraph/lib => github.com/sourcegraph/sourcegraph-public-snapshot/lib v0.0.0-20240822153003-c864f15af264
+	github.com/sourcegraph/sourcegraph/conc => github.com/sourcegraph/sourcegraph-public-snapshot/conc v0.0.0-20240822153003-c864f15af264
```
Comment (jmank88) [inline deployment/go.mod]: What is `/conc`?

Comment (cgruber) [reply deployment/go.mod]: I have no idea. It comes from //indirect.  I assume it's maybe something concurrency. We're not depending on it, go mod tidy adds it.

Comment (jmank88) [reply deployment/go.mod]: But it doesn't exist? https://github.com/sourcegraph/sourcegraph-public-snapshot/tree/c864f15af264f0f456a6d8a83290b5c940715349

Comment (cgruber) [reply deployment/go.mod]: https://github.com/sourcegraph/conc

Comment (cgruber) [reply deployment/go.mod]: I misread it. Fixing.

Comment (jmank88) [reply deployment/go.mod]: Oh well there is an extra `/sourcegraph` in here:
```suggestion
	github.com/sourcegraph/conc => github.com/sourcegraph/sourcegraph-public-snapshot/conc v0.0.0-20240822153003-c864f15af264
```

Comment (jmank88) [reply deployment/go.mod]: But that repo still exists, so we shouldn't need a replace :shrug: 
https://github.com/sourcegraph/conc
Can we drop this completely?

Comment (cgruber) [reply deployment/go.mod]: I just did. This was me cargo-culting the prior lib redirect, because I misread the conc one. Removed.

**Review by jmank88 — APPROVED** [review]

**Comment by cgruber** [comment]
The commit message is wrong now, but it's sort of in the right zone. lol. I can fix it if it matters.

**Review by connorwstein — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**